src/condor_dagman/parse.cpp

```diff
@@ -1745,8 +1745,27 @@ static bool parse_vars(Dag *dag, const char *filename, int lineNumber)
 
 	std::string varName;
 	std::string varValue;
-
-	char *varsStr = strtok( NULL, "\n" ); // just get all the rest -- we'll be doing this by hand
+	bool prepend; //Bool for if variable is prepended or appended. Default is false
+
+	char *varsStr = strtok( NULL, "\n" ); // just get all the rest and we will manually parse vars
+
+/*	
+*	Check to see if PREPEND or APPEND was specified before variables to be passed.
+*	If option is found then we set prepend boolean appropriately and increment
+*	the original varsStr pointer to point at char after option. -Cole Bollig
+*/
+
+        if (starts_with(varsStr,"PREPEND") ){//See if PREPEND is at beginning of token
+		prepend = true;
+		varsStr += 7;
+	} else if ( starts_with(varsStr,"APPEND") ){//See if APPEND is at beginning of token
+		prepend = false;
+		varsStr += 6;
+	} else {
+		//If options aren't found then set to global knob
+		// !append -> prepend
+		prepend = !param_boolean("DAGMAN_DEFAULT_APPEND_VARS", false);
```

Comment:
Isn't this param_boolean call being made when we init the dag? why not use that value?